bindings/src/Capgemini.PowerApps.SpecFlowBindings/Configuration/TestConfiguration.cs

```diff
@@ -69,7 +73,24 @@ public UserConfiguration GetUser(string userAlias)
         {
             try
             {
-                return this.Users.First(u => u.Alias == userAlias);
+                if (this.Users.Count > 1)
+                {
+                    if (this.Users.Count - 1 == this.aliasIndex)
+                    {
+                        this.aliasIndex = 0;
+                    }
+                    else
+                    {
+                        this.currentUser = this.Users[this.aliasIndex];
+                        this.aliasIndex++;
+                    }
+
+                    return this.currentUser;
+                }
+                else
+                {
+                    return this.Users.First(u => u.Alias == userAlias);
+                }
```

Comment:
So this will technically work but only assuming that all of your user aliases are the same which won't generally be the case. What we need to do is maintain a current index across all of the user aliases. Perhaps something like:

```suggestion
                var aliasEnumerator = this.userEnumerators[userAlias];

                if (!aliasEnumerator.MoveNext())
                {
                    aliasEnumerator.Reset();
                    aliasEnumerator.MoveNext();
                }

                return aliasEnumerator.Current;        
```